extensions/assets-manager/AssetsManagerEx.h

```diff
@@ -67,16 +67,33 @@ class CC_EX_DLL AssetsManagerEx : public Ref
         FAIL_TO_UPDATE
     };
     
+    struct Config {
+        /** @brief  Function used to calculate the storage path of each asset
+         */
+        std::function<std::string(const std::string& key, const Manifest::Asset& asset)> getAssetStoragePath;
+
+        /** @brief  Hints for configuring downloader
+         */
+        network::DownloaderHints downloaderHints;
+
+        /** @brief  Default configuration for AssetsManagerEx
+         */
+        static const Config& defaultConfig();
+    };
+
     const static std::string VERSION_ID;
     const static std::string MANIFEST_ID;
     
     /** @brief Create function for creating a new AssetsManagerEx
      @param manifestUrl   The url for the local manifest file
      @param storagePath   The storage path for downloaded assets
+     @param config        AssetsManagerEx configuartion
      @warning   The cached manifest in your storage path have higher priority and will be searched first,
                 only if it doesn't exist, AssetsManagerEx will use the given manifestUrl.
      */
-    static AssetsManagerEx* create(const std::string &manifestUrl, const std::string &storagePath);
+    static AssetsManagerEx* create(const std::string &manifestUrl,
```

Comment:
It is better not use default parameter to obey [cocos coding style](https://github.com/cocos2d/cocos2d-x/blob/v3/docs/CODING_STYLE.md#default-arguments).